x-pack/solutions/security/plugins/security_solution/server/lib/siem_migrations/rules/data/rule_migrations_data_integrations_client.ts

```diff
@@ -28,32 +23,52 @@ export class RuleMigrationsDataIntegrationsClient extends RuleMigrationsDataBase
   }
 
   /** Indexes an array of integrations to be used with ELSER semantic search queries */
-  async create(): Promise<void> {
+  async populate(): Promise<void> {
     const index = await this.getIndexName();
-    await this.esClient
-      .bulk(
-        {
-          refresh: 'wait_for',
-          operations: INTEGRATIONS.flatMap((integration) => [
-            { update: { _index: index, _id: integration.id } },
-            {
-              doc: {
-                title: integration.title,
-                description: integration.description,
-                data_streams: integration.data_streams,
-                elser_embedding: integration.elser_embedding,
-                '@timestamp': new Date().toISOString(),
+    const packages = await this.dependencies.packageService?.asInternalUser.getPackages();
+    if (packages) {
+      const ragIntegrations = packages.map((pkg) => ({
+        title: pkg.title,
+        id: pkg.name,
+        description: pkg.description,
+        data_streams:
+          pkg.data_streams?.map((stream) => ({
+            dataset: stream.dataset,
+            index_pattern: `logs-${stream.dataset}-*`,
+            title: stream.title,
+          })) || [],
+        elser_embedding: [
+          pkg.title,
+          pkg.description,
+          ...(pkg.data_streams?.map((stream) => stream.title) || []),
+        ].join(' - '),
+      })) as RuleMigrationIntegration[];
+      await this.esClient
+        .bulk(
+          {
+            refresh: 'wait_for',
+            operations: ragIntegrations.flatMap((integration) => [
+              { update: { _index: index, _id: integration.id } },
+              {
+                doc: {
+                  title: integration.title,
+                  description: integration.description,
+                  data_streams: integration.data_streams,
+                  elser_embedding: integration.elser_embedding,
+                },
+                doc_as_upsert: true,
               },
-              doc_as_upsert: true,
-            },
-          ]),
-        },
-        { requestTimeout: 10 * 60 * 1000 }
-      )
-      .catch((error) => {
-        this.logger.error(`Error preparing integrations for SIEM migration ${error.message}`);
-        throw error;
-      });
+            ]),
+          },
+          { requestTimeout: 10 * 60 * 1000 }
+        )
+        .catch((error) => {
+          this.logger.error(`Error preparing integrations for SIEM migration ${error.message}`);
+          throw error;
+        });
+    } else {
+      this.logger.error('Error retrieving packages for SIEM migration');
```

Comment:
This is not an error strictly speaking, could we do this instead:
```suggestion
      this.logger.warn('Package service not available, not able not populate integrations index')
```